### ISSUE TITLE
XHR interceptor not respecting axios interceptor

### ISSUE THREAD

**Issue by rishabhpoddar** [supertokens/supertokens-website#208]
Seems like when a user adds an axios request interceptor, it runs after our XHR interceptor which means that if they modify the axios request headers in their interceptor, it gives the following error:
```
Uncaught DOMException: Failed to execute 'setRequestHeader' on 'XMLHttpRequest': The object's state must be OPENED.
```

For now, the work around is to disable the XHR interceptor that supertokens has (via override), and add the axios interceptor using the `addAxiosInterceptor` function from supertokens.

**Comment by komarovalexander** [comment]
Hi @rishabhpoddar looks like I faced with that issue 

`the work around is to disable the XHR interceptor that supertokens has (via override), and add the axios interceptor using the addAxiosInterceptor function from supertokens.`

could you give an example or doc to the link how to do that?

**Comment by rishabhpoddar** [comment]
hey @komarovalexander can you tell us how to replicate this issue?

The work around is as follows:

Disable XHR interception on the frontend:
```
Session.init({
    override: {
        functions: (oI) => {
            return {
                ...oI,
                addXMLHttpRequestInterceptor: (input) => {
                    
                }
            }
        }
    }
})
```

Add axios interceptor to your axios instance:

```
import {addAxiosInterceptors} from "supertokens-web-js/recipe/session"

addAxiosInterceptors(axios)
```

If you are using supertokens-auth-react, import from that instead of supertokens-web-js

**Comment by komarovalexander** [comment]
Hi @rishabhpoddar we just updated server side version and now we have staled token and infinite refresh loop. we use supertokens-auth-react on UI side

**Comment by rishabhpoddar** [comment]
Can you give us more info?
- Frontend and backend debug logs
- Refresh API request and response headers
- The request and response headers for the API that you are trying to call which gives back a 401.

**Comment by vivek1996** [comment]
Hi @rishabhpoddar,
 Scenario to replicate  _XHR interceptor not respecting axios interceptor_ issue:

The `axiosInstance` in our application is configured with an interceptor that adds an `Authorization` header to the API request using the token obtained from the `getAccessToken` method. This works fine during login, but the API requests do not have the `Authorization` header after reloading the page. which results in 401 & getting logged out.

Upon inspecting the `axios interceptor` using console logs, I can see that the token is successfully obtained from the `getAccessToken` method and is being returned as part of the `config` object.

Our application's authentication provider is set as `active-directory`.

```javascript
 const getAccessToken = async () => {
    if (await Session.doesSessionExist()) {
        return (await Session.getAccessTokenPayloadSecurely()).jwt;
    }
};

axiosInstance.interceptors.request.use(async (config) => {
         const token = await getAccessToken();
        config.headers.Authorization = `Bearer ${token}`;
        return config;
    });
   ```

**Comment by rishabhpoddar** [comment]
hey @vivek1996 we are still not able to reproduce the issue on our end, even when using the code snippet above. Do you mind showing us over a call please? You can pick a time here: https://supertokens.com/call-user

Thanks.

**Comment by rishabhpoddar** [comment]
This has been fixed in version ?= 16.0.6
